src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,223 @@
+ */
+public class Parser {
+    /**
+     * Checks if dateString is in a valid date form depicted by dateFormatter.
+     *
+     * @param dateString String which contains a date.
+     * @param dateFormatter Format which you want to check if dateString is in.
+     * @return true if dateString is in valid form, else otherwise.
+     */
+    public static boolean isValidDate(String dateString, DateTimeFormatter dateFormatter) {
+        try {
+            LocalDate.parse(dateString, dateFormatter);
+        } catch (DateTimeParseException e) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Checks if timeString is in a valid date form depicted by timeFormatter.
+     *
+     * @param timeString String which contains a time.
+     * @param timeFormatter Format which you want to check if timeString is in.
+     * @return true if timeString is in valid form, else otherwise.
+     */
+    public static boolean isValidTime(String timeString, DateTimeFormatter timeFormatter) {
+        try {
+            LocalTime.parse(timeString, timeFormatter);
+        } catch (DateTimeParseException e) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Checks if input is a remove command.
+     *
+     * @param input Input which is being checked.
+     * @return true if input is a remove command.
+     */
+    public static boolean isRemove(String input){
+        return input.startsWith("remove ");
+    }
+
+    /**
+     * Checks if input is a done command.
+     *
+     * @param input Input which is being checked.
+     * @return true if input is a done command.
+     */
+    public static boolean isDone(String input) {
+//        String[] splited = phrase.split(" ");
+////        System.out.println(splited);
+//        if (splited.length != 2) {
+//            return false;
+//        } else {
+//            return splited[0].equals("done") && splited[1]
+//                  .matches("\\d+") && Integer.valueOf(splited[1]) <= listLength;
+//        }
+        return input.startsWith("done ");
+    }
+
+    /**
+     * Checks if user input is a find command.
+     *
+     * @param input User input to check if it is a find command.
+     * @return true if it is a find command, else false.
+     */
+    public static boolean isFind(String input) {
+        return input.startsWith("find ");
+    }
+
+    /**
+     * Splits string which contains a description and possibly a deadline of Task Objects.
+     *
+     * @param input Input which contains a description and possibly a deadline.
+     * @param type Type of Task information the input contains.
+     * @return String[] containing the description at index 0 and the deadline (if any) at index 1.
+     * @throws DukeException If no deadline or description for respective Task objects.
+     */
+    public static String[] splitInput(String input, String type) throws DukeException {
```

Comment:
I like how the logic of the code was group neatly. Should there also be comments to better explain what the groups of code is trying to achieve ?